src/Symfony/Bundle/FrameworkBundle/Templating/TemplateResponse.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Templating;
+
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @author Pierre du Plessis <pdples@gmail.com>
+ */
+class TemplateResponse implements TemplateResponseInterface
```

Comment:
You could simply return a new JsonResponse in that case like it's done now. It doesn't really make sense to me to first create an object to tell a listener to then create the Response for something you could've done in your controller already, this would trigger the view listeners which is just overhead.